intrinsic_and_extrinsic_optimise_real_data/src/camera_calibration_real_data.cpp

```diff
@@ -0,0 +1,422 @@
+	{
+		return (new ceres::AutoDiffCostFunction<ReProjectionResidual, 2, 6, 4>(
+			new ReProjectionResidual(pixel_points, board_T_point)));
+	}
+
+	// Declare struct variables
+	private:
+		double observed_pixel_points_[3];
+		double board_T_point_[4];
+};
+
+int main()
+{
+	unsigned int count = 0;
+	unsigned int num_of_images = 0;
+	unsigned int total_chessboards_detected = 0;
+	cv::VideoCapture cap("../Calibration_images_numbered/image%02d.jpg");
+
+	cv::Size patternsize = cv::Size(7, 6); // (width, height) or (columns, rows) or (X, Y)
+	std::vector<cv::Mat> images;
+	std::vector<std::vector<cv::Point2f>> Chessboard_corners;
+
+	// This loop goes into the folder defined previously and reads the images one by one.
+	// The files MUST be of the same type and have the same name.
+	// The files MUST have the same name and have a consecutive numbering scheme
+	// For example: image01.jpg, image02.jpg, image03.jpg ...
+	while(cap.isOpened())
+	{
+		// The image is read into the OpenCV matrix vector here
+		cv::Mat img;
+		cap.read(img);
+		if(img.empty())
+		{
+			std::cout << "End of Sequence \r\n";
+			break;
+		}
+		
+		// The recently read image is processed here
+		std::vector<cv::Point2f> corners; //This will be filled by the detected corners
+		bool chessboard_found = cv::findChessboardCorners(img, patternsize, corners);
+
+		if(chessboard_found)
+		{
+			std::cout << "The Chessboard was found within the image: " << num_of_images << "\r\n";
+			total_chessboards_detected++;
+			images.push_back(img); // Only store the images where the Chessboards were found.
+			Chessboard_corners.push_back(corners); // Store the calibration points from the image but only for the chessboards that were found.
+
+			// cv::namedWindow("Display window");
+			// cv::drawChessboardCorners(img, patternsize, corners, chessboard_found);
+			// cv::imshow("Dsplay Window", img);
+			// cv::waitKey(0);
+		}
+		else
+		{
+			std::cout << "The Chessboard was not found within the image: " << num_of_images << "\r\n";
+		}
+		num_of_images++;
+	}
+
+	std::cout << "Finished reading from the folder! \r\n";
+	std::cout << "The total number of images found: " << num_of_images << "\r\n";
+	std::cout << "The number of images with Chess boards within the image: " << total_chessboards_detected << "\r\n\n";
+
+	// The calibration board is generated here and is regarded as matrix D
+	std::vector<cv::Point3f> matrix_D;
+ 
+	for (int y = 0; y != Number_of_internal_corners_y; y++)
+	{
+		for (int x = 0; x != Number_of_internal_corners_x; x++)
+		{
+			matrix_D.push_back(cv::Point3f(x * Block_dimension, y * Block_dimension, 0 * Block_dimension));
+		}
+	}
+
+	// The solvePnP function is be used here to determine the initial estimates for the camera extrinsics for each image.
+	double matrix_B[3][3] = {{initial_fx, 0, initial_cx}, {0, initial_fy, initial_cy}, {0, 0, 1} };
+	cv::Mat matrix_B_ = cv::Mat(3, 3, CV_64FC1, matrix_B);
```

Comment:
Save underscores for member variables.
https://google.github.io/styleguide/cppguide.html#Variable_Names